### PR TITLE
[#45875] Change API response of storage files endpoint

### PR DESCRIPTION
**Pull request by apfohl**
https://community.openproject.org/work_packages/45875

### CODE REVIEW

**Comment by Kharonus** [comment]
first question first: The update to the API specification comes in a later PR? @apfohl ?

**Comment by apfohl** [comment]
I moved this to another work package: https://community.openproject.org/projects/openproject/work_packages/45954/activity

File: frontend/src/app/core/state/storage-files/storage-files.model.ts [hunk 1]
```diff
@@ -26,7 +26,12 @@
...
-import { QueryEntity } from '@datorama/akita';
-import { StoragesState } from 'core-app/core/state/storages/storages.store';
+import { IHalResourceLinks } from 'core-app/core/state/hal-resource';
+import { IStorageFile } from 'core-app/core/state/storage-files/storage-file.model';
 
-export class StoragesQuery extends QueryEntity<StoragesState> {}
+export interface IStorageFiles {
+  files:IStorageFile[];
+  parent:IStorageFile;
+  _type:string;
```
Comment (Kharonus) [inline frontend/src/app/core/state/storage-files/storage-files.model.ts]: minor: `_type` is of a literal type, hence I would replace `string` with `'StorageFiles'`, or whatever you defined in the API specifications.

**Review by Kharonus — APPROVED** [review]
That looks very fine and complete from my POV. The minor thing with the literal type can be fixed, I'd like it to, but this is no blocker for the approval.

**Pull request merged**